jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletContextHandler.java

```diff
@@ -2711,87 +2711,10 @@ public ContextHandler getContextHandler()
-                {
-                    // look first for vhost matching context only
-                    if (getVirtualHosts() != null && getVirtualHosts().size() > 0)
-                    {
-                        if (ch.getVirtualHosts() != null && ch.getVirtualHosts().size() > 0)
-                        {
-                            for (String h1 : getVirtualHosts())
-                            {
-                                for (String h2 : ch.getVirtualHosts())
-                                {
-                                    if (h1.equals(h2))
-                                    {
-                                        if (matchedPath == null || contextPath.length() > matchedPath.length())
-                                        {
-                                            contexts.clear();
-                                            matchedPath = contextPath;
-                                        }
-
-                                        if (matchedPath.equals(contextPath))
-                                            contexts.add(ch);
-                                    }
-                                }
-                            }
-                        }
-                    }
-                    else
-                    {
-                        if (matchedPath == null || contextPath.length() > matchedPath.length())
-                        {
-                            contexts.clear();
-                            matchedPath = contextPath;
-                        }
-
-                        if (matchedPath.equals(contextPath))
-                            contexts.add(ch);
-                    }
-                }
-            }
-
-            if (contexts.size() > 0)
-                return contexts.get(0).getServletContext();
-
-            // try again ignoring virtual hosts
-            matchedPath = null;
-            for (ServletContextHandler ch : handlers)
-            {
-                if (ch == null)
-                    continue;
-                String contextPath = ch.getContextPath();
-
-                if (uripath.equals(contextPath) || (uripath.startsWith(contextPath) && uripath.charAt(contextPath.length()) == '/') || "/".equals(contextPath))
-                {
-                    if (matchedPath == null || contextPath.length() > matchedPath.length())
-                    {
-                        contexts.clear();
-                        matchedPath = contextPath;
-                    }
-
-                    if (matchedPath.equals(contextPath))
-                        contexts.add(ch);
-                }
-            }
-
-            if (contexts.size() > 0)
-                return contexts.get(0).getServletContext();
+            //TODO jetty-12 does not currently support cross context dispatch
+            String myPath = getContextPath();
+            if (myPath != null && myPath.equals(uripath))
+                return this;
```

Comment:
I think we should always return null as we have suggested that `servletContext.getContext(servletContext.getContextPath())==null` is a good test for if cross context dispatch is supported or not.

Reply:
I did it the same way that ee9 is handling `getContext(String)`, which is to return the current context if there's a match, see: https://github.com/eclipse/jetty.project/blob/jetty-12.0.x/jetty-ee9/jetty-ee9-nested/src/main/java/org/eclipse/jetty/ee9/nested/ContextHandler.java#L1750

Do you want me to change both ee9 and ee10 to return null?

Reply:
yes please.  This way the TCK tests can simply check if it is supported by trying to lookup themselves